index.html

```diff
@@ -34,32 +34,77 @@
+            promises.push(itowns.Fetcher.json('examples/layers/JSONLayers/WORLD_DTM.json').then(function (result) { return globeView.addLayer(result); }));
+            promises.push(itowns.Fetcher.json('examples/layers/JSONLayers/IGN_MNT_HIGHRES.json').then(function (result) { return globeView.addLayer(result); }));
 
             menuGlobe.addGUI('RealisticLighting', false,
-                function(newValue) { globeView.setRealisticLightingOn(newValue); });
+                function (newValue) { globeView.setRealisticLightingOn(newValue); });
 
             Promise.all(promises).then(function () {
                 menuGlobe.addImageryLayersGUI(globeView.getLayers(function (l) { return l.type === 'color'; }));
-                menuGlobe.addElevationLayersGUI(globeView.getLayers(function(l) { return l.type === 'elevation'; }));
+                menuGlobe.addElevationLayersGUI(globeView.getLayers(function (l) { return l.type === 'elevation'; }));
                 console.info('menuGlobe initialized');
-            }).catch( function (e) { console.error(e) });
+
+                var urlKml = 'https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/croquis.kml';
+                var urlGpx = 'https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/ULTRA2009.gpx';
+                var urlGeojson = 'https://raw.githubusercontent.com/gregoiredavid/france-geojson/master/departements/09-ariege/departement-09-ariege.geojson';
+
+                itowns.Fetcher.xml(urlKml).then(function (file) {
+                    const kmlLayer = {
+                        type: 'color',
+                        file,
+                        protocol: 'vectortocolor',
+                        projection: 'EPSG:4326',
+                        id: 'Kml',
+                    };
+                    globeView.addLayer(kmlLayer);
+                    menuGlobe.addImageryLayersGUI(globeView.getLayers(function (l) { return l.id === 'Kml'; }));
+                });
+
+                itowns.Fetcher.xml(urlGpx).then(function (file) {
+                    const gpxLayer = {
+                        type: 'color',
+                        file,
+                        protocol: 'vectortocolor',
+                        projection: 'EPSG:4326',
+                        id: 'Gpx',
+                    };
+                    globeView.addLayer(gpxLayer);
+                    menuGlobe.addImageryLayersGUI(globeView.getLayers(function (l) { return l.id === 'Gpx'; }));
+                });
+
+                itowns.Fetcher.json(urlGeojson).then(function (file) {
+                    const geojsonLayer = {
+                        type: 'color',
+                        file,
+                        protocol: 'vectortocolor',
+                        id: 'ariege',
+                        projection: 'EPSG:4326',
+                        crsFile: 'EPSG:4326',
```

Comment:
`crsFile` looks like a duplicate of `projection`. Are they different?

Reply:
Yes I'm not sure about this.
It is more the projection of destination,
I can rename it to `crsOut` 
or automatically assign it when adding the layer in the `addLayer` function

Reply:
`projection` should keep it's current meaning = `crs` of the data of this layer.
I don't think you need to introduce another property...

```js
    layer.geojson = GeoJSON2Three.parse(layer.projection, layer.file, layer.extent, options); 
```
is wrong, because the first parameter should be the attached geometry crs.

```js
options.crsIn = options.crsIn || readCRS(json);
```
Should use layer.projection instead.

IMHO you should try to add gpx file to the Lyon example to see it works properly.

Reply:
I push a proposition https://github.com/iTowns/itowns/pull/450/commits/25fd9556ef713c0487135d3391f25c47269a6a18

Reply:
with gpx test in planar mode